uber/models.py

```diff
@@ -1625,6 +1817,464 @@ def must_contact(self):
+            consisting entirely of whitespace.
+        discount (int): The discount amount that should be applied to the
+            purchase price of a badge. The interpretation of this value
+            depends on the value of `discount_type`. In any case, a value of
+            0 equates to a full discount, i.e. a free badge.
+        discount_display (str): A human readable description of the discount.
+        discount_type (int): The type of discount this promo code will apply.
+            Valid values are:
+
+            * 0 `FIXED_DISCOUNT`: `discount` is interpreted as a fixed
+                dollar amount by which the badge price should be reduced. If
+                `discount` is 49 and the badge price is normally $100, then
+                the discounted badge price would be $51.
+
+            * 1 `FIXED_PRICE`: `discount` is interpreted as the actual badge
+                price. If `discount` is 49, then the discounted badge price
+                would be $49.
+
+            * 2 `PERCENT_DISCOUNT`: `discount` is interpreted as a percentage
+                by which the badge price should be reduced. If `discount` is
+                20 and the badge price is normally $50, then the discounted
+                badge price would $40 ($50 reduced by 20%). If `discount` is
+                100, then the price would be 100% off, i.e. a free badge.
+
+        expiration_date (datetime): The date & time upon which this promo code
+            expires. An expired promo code may no longer be used to receive
+            discounted badges.
+        is_expired (bool): True if this promo code is expired, False otherwise.
+        is_unlimited (bool): True if this promo code may be used an unlimited
+            number of times, False otherwise.
+        is_valid (bool): True if this promo code is still valid and may be
+            used again, False otherwise.
+        normalized_code (str): A normalized version of `code` suitable for
+            database queries. Normalization converts `code` to all lowercase
+            and removes dashes ("-").
+        used_by (list): List of attendees that have used this promo code.
+            Note:
+                This property is declared as a backref in the Attendee class.
+        uses_allowed (int): The total number of times this promo code may be
+            used. A value of None means this promo code may be used an
+            unlimited number of times.
+        uses_allowed_display (int): A human readable description of
+            uses_allowed.
+        uses_count (int): The number of times this promo code has already
+            been used.
+        uses_count_display (int): A human readable description of uses_count.
+        uses_remaining (int): Remaining number of times this promo code may
+            be used.
+        uses_remaining_display (int): A human readable description of
+            uses_remaining.
+    """
+
+    FIXED_DISCOUNT = 0
```

Comment:
You're totally right that we have a bunch of config living in our config files that shouldn't be there! This actually gets at one of the core problems we have right now with the system, which is that we have three kinds of 'config':
1. Config like the stuff you define here, which are literally just static enumerations that we want to access as global vars. The treatment of `enums` in our configspec.ini (which itself is defined in config.py) helps us ensure that when we add new stuff that code depends upon, it's handled in a consistent manner.
2. Actual config options we want to change sometimes, like dates and badge prices.
3. "Dynamic" config options that are usually some permutation of the above two -- all these are defined in config.py.

I'd actually argue that configspec.ini and development-defaults.ini is where type 1 config belongs, and it's type 2 config options that we want to move out of there into other parts of the system. Eventually, ideally, we would not be changing our flat-file config the way we do now.

In any case, the core problem is that these _look_ like existing config options but do not behave like them, particularly, they are not globally accessible by our templates and there is no `c.DISCOUNTS` to accompany `c.DISCOUNT_OPTS` (well, plus `c.` wouldn't even be used). This will definitely cause confusion down the line. If we really don't care about them being globally accessible via our `c` config object, we should at least not use the global variable styling of all caps.

We should totally continue this discussion in Slack. :)